WordPress/gutenberg
# Sidebar: Update appearance of active items

Closes https://github.com/WordPress/gutenberg/issues/66554.

## What / Why?
This PR updates the appearance of active menu items in the navigation sidebar to achieve two objectives:

1. Remove appearance overlap with primary buttons
2. Better manage prominence (there's no need for the active menu item to command such a strong appearance)

## How
- Removed background on hover state
- Apply `$gray-800` background to active state
- Increased font weight on active state

The increased font-weight is an a11y consideration, due to the contrast between the sidebar and active item backgrounds not being sufficient. 

## Testing Instructions
1. Open the Site Editor
2. Click around in the Sidebar
4. Observe updated styling for hover / active states.

## Before

https://github.com/user-attachments/assets/8dc2a5dc-bb0a-4185-a0e2-8f251596140d



## After

https://github.com/user-attachments/assets/2bd1f031-7a4a-4a34-b463-85e172408bf6
